pkg/transport/entry.go

```diff
@@ -47,20 +48,23 @@ type Entry struct {
 }
 
 // MakeEntry creates a new transport entry
-func MakeEntry(pk1, pk2 cipher.PubKey, tpType string, public bool, label Label) Entry {
-	return Entry{
-		ID:     MakeTransportID(pk1, pk2, tpType),
-		Edges:  SortEdges(pk1, pk2),
+func MakeEntry(initiator, target cipher.PubKey, tpType string, public bool, label Label) Entry {
+	entry := Entry{
+		ID:     MakeTransportID(initiator, target, tpType),
```

Comment:
If I read this correctly, we may well end up with a scenario where we have two entries with the same ID, given that we still sort by  PubKey value for generating IDs but create entries by initiator. So if two visors established transports to each other, the ID would be the same. 
1. May that create trouble getting transports by ID from transport discovery? 
2. Are we currently checking (and preventing) a transport to be created if a transport of the same type between the same two edges already exists?

Reply:
Yes, it is possible to create two entries with the same ID but different ordering of edges, but since IDs are used to discriminate entries it shouldn't matter. 

1. I don't think so. ID is the same for entries (a, b) and (b, a). Transport discovery uses ID to store the entry, and if there is such ID, it does nothing. Other parts of TD just use the ID.

2. Yes, upon startup visor gets all transports from transport discovery and saves them internally. When there is a request to establish new transport, it first checks if transport already exists, using transport ID to determine it.

I can imagine a scenario when two visors try to establish a transport simultaneously to each other, and each holding its own version of entry. There will be no double connection: when visor accepts a new connection that corresponds to an existing transport it will automatically close one of them (this time using smallest PK logic). They will probably both try to redial when the connection is lost since both think they are the initiator, and probably will close one of the connections when there are two for one transport after redial. This is speculating from the code, I haven't performed any tests.

This sounds unlikely though since two visors should create a transport at the same time.